velox/serializers/PrestoSerializer.cpp

```diff
@@ -2008,11 +1968,14 @@ void PrestoVectorSerde::deserialize(
   const bool useLosslessTimestamp = prestoOptions.useLosslessTimestamp;
   auto codec = common::compressionKindToCodec(prestoOptions.compressionKind);
   auto numRows = source->read<int32_t>();
-  if (!(*result) || !result->unique() || (*result)->type() != type) {
+
+  if (*result && result->unique()) {
+    VELOX_CHECK((*result)->type() == type);
+    (*result)->prepareForReuse();
```

Comment:
nit: Would it be lot of additional work to change deserialize api to take shared_ptr<RowVector>& result ?

Reply:
This is doable, but probably not in this PR.